### ISSUE TITLE
Check if bootWar/war tasks are enabled

### ISSUE THREAD

**Issue by mattysweeps** [GoogleContainerTools/jib#2096]
**Environment**:

- *Jib version:* 1.7.0
- *Build tool:* gradle 5.2.1
- *OS:* linux

**Description of the issue**:
When a project enables the war task but disables the bootWar task, Jib incorrectly chooses the bootWar task to depend on, instead of the war task.

**Expected behavior**:
Jib should depend on enabled tasks only.

**Steps to reproduce**:

  1. apply plugins 'org.springframework.boot' and jib
  2. set `bootWar { enabled = false }` and `war { enabled = true }` in build.gradle
  3. gradle jib

**Log output**: 

**Additional Information**: 
https://github.com/GoogleContainerTools/jib/blob/4d8efc902d40945fe5ed296d3bf55f0d42119fef/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/JibPlugin.java#L167

https://github.com/GoogleContainerTools/jib/blob/4d8efc902d40945fe5ed296d3bf55f0d42119fef/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TaskCommon.java#L49

**Comment by loosebazooka** [comment]
This seems like the right thing to do, but out of curiosity: why are you disabling the bootwar task?

**Comment by mattysweeps** [comment]
@loosebazooka The project I'm trying to use with Jib is github.com/cloudfoundry/uaa, specifically the `./uaa/build.gradle` https://github.com/cloudfoundry/uaa/blob/develop/uaa/build.gradle

I'm not sure why bootWar is disabled, but I believe it's because the app is transitioning to a spring boot app from a spring XML one.

Here's my PR for the changes I made https://github.com/cloudfoundry/uaa/pull/1129

**Comment by chanseokoh** [comment]
Hey @slowr, asking for help, since you've recently modified our code to use `TaskProvider`. It seems like in order to determine if the bootWar task is enabled, I need to retrieve it first (i.e., `.get().getEnabled()`). And then Jib should dynamically determine whether it should depend on the war task or the bootWar task. But calling `TaskProvider.get()` defeats #2072. What should be the best way?

**Comment by slowr** [comment]
I don't think it is possible to get the enabled state without creating the task. I asked the Gradle community and this was the answer I got back:
```
probably not, because enable/disable can be  written to by configuration actions attached to the provider, so you wouldn't get the right answer. in fact, I think all tasks would report enabled = true
```

**Comment by loosebazooka** [comment]
Can we depend on both as task dependencies and query the correct config data later?

**Comment by chanseokoh** [comment]
Ah, yeah, haven't thought about it. I think it can be the best compromise. That is, we configure Jib runs to depend on both war and bootWar tasks if they are available.

**Comment by chanseokoh** [comment]
@mattysweeps I realized the reproducible steps you wrote aren't really correct, so let me check if we are on the same page.

> 1. apply plugins 'org.springframework.boot' and jib
> 2. set `bootWar { enabled = false }` in build.gradle
> 3. gradle jib

AFAIK, when you apply 'org.springframework.boot' to a WAR project (i.e., a project where the War plugin is applied), it automatically creates the `bootWar` task and disables the `war` task. This is [documented]( https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/html/#packaging-executable-and-normal), and from my experiment this seems true. Then, if you just disable `bootWar`, both `war` and `bootWar` are disabled and no WAR file is ever produced. So, there should be an additional step to set `war { enabled = true }`. Did I miss something?

**Comment by mattysweeps** [comment]
@chanseokoh Ah, I see. I believe you are correct. That's what I see in the `build.gradle` file I'm using:
https://github.com/cloudfoundry/uaa/blob/develop/uaa/build.gradle#L16

I'll update the issue description

**Comment by chanseokoh** [comment]
@mattysweeps we just released 1.8.0 with the fix.
